hw/ip/otbn/data/bignum-insns.yml

```
@@ -453,21 +453,27 @@
   synopsis: Pseudo-modulo subtraction
   operands: [wrd, wrs1, wrs2]
   doc: |
-    Subtracts the second WDR value from the first WDR value, performs a modulo operation with the MOD WSR, and writes the result to the destination WDR.
-    This operation is equivalent to a modulo subtraction as long as `wrs1 - wrs2 >= -MOD` holds.
-    This constraint is not checked in hardware.
+    Subtract `<wrs2>` from `<wrs1>`, modulo the `MOD` WSR.
+
+    The intermediate result is treated as a signed number (of width `WLEN + 1`).
+    If it is negative, `MOD` is added to it.
+    The 2's-complement result is then truncated to 256 bits and stored in `<wrd>`.
+
+    This operation correctly implements subtraction modulo `MOD`, providing that the intermediate result is not less than `-MOD`.
```

I think this sentence is not entirely correct. I see that this also wasn't accurate in the original definition and since we are at it, I suggest to fix it here.
When we say it correctly implements "subtraction modulo MOD" we say that it implements c = a-b mod `MOD`. Here, this is only correct if the intermediate result is greater or equal `-MOD` but also less than `MOD`.
Say MOD=3 and a=7 and b=2, result would be c=5 (and that is how it will be implemented in OTBN), but from the definition here, one would expected that the result would be c=2
```suggestion
    This operation correctly implements subtraction modulo `MOD`, providing that the intermediate result is in the range of `-MOD` to `MOD-1`.
```
Of course the next sentence is a sufficiency for this but not a necessity. Say `MOD`=3 and a=4 and b=2, result would be c=2 and correct even though not both operands are smaller than `MOD`.

---

Ooh, good point! Will fix.